### ISSUE TITLE
Feature Request : Visual Pattern Editor

### ISSUE THREAD

**Issue by MRSEEO** [rbecker5/BulletUpHell#4]
**What is your use-case ?**
I wanna make a different patterns for my bullethell game, but starting a scene after every change is... long

**Describe the feature you'd like**
Scene with menu for pattern editing and testing. Also button for saving. But i idk if it possible...

**Comment by Dark-Peace** [comment]
You mean preview the bullets trajectories and things like that ? Yeah I wish I could. With the amount of features to take into account, it would be a gigantic task. Another possibility would be to allow to shoot inside the editor but then half of the features, like collisions, homing,... wouldn't work in the editor.

My issue is that not everything is previewable. Maybe if you tell me more precisely what you'd like to see previewed, I can do that alone ? I can't promise anything, and especially regarding when it'd be done.

**Comment by MRSEEO** [comment]
> You mean preview the bullets trajectories and things like that ? Yeah I wish I could. With the amount of features to take into account, it would be a gigantic task. Another possibility would be to allow to shoot inside the editor but then half of the features, like collisions, homing,... wouldn't work in the editor.
> 
> My issue is that not everything is previewable. Maybe if you tell me more precisely what you'd like to see previewed, I can do that alone ? I can't promise anything, and especially regarding when it'd be done.

Yeah, i mean preview the bullets trajectories. Like this:
![image](https://github.com/Dark-Peace/BulletUpHell/assets/51641515/0090fb70-42bf-4c75-8cbd-8186466fd75e)
![image](https://github.com/Dark-Peace/BulletUpHell/assets/51641515/7e5b3a05-bdbb-439b-9612-bf8ff9ae2ac7)

(it's from BulletHellper for Godot 3.5 and PatternMaker runs in a special game scene)

This function is not necessary, but i would like to see something like it...
Anyway thank you SO MUCH for this awesome addon!!!

**Comment by Dark-Peace** [comment]
Yeah, I ruled out this feature 'cause there was no way I could make it compatible with every pattern and bullet property. But maybe some basic things could be added like spawn position and shoot direction.

**Comment by MRSEEO** [comment]
> Yeah, I ruled out this feature 'cause there was no way I could make it compatible with every pattern and bullet property. But maybe some basic things could be added like spawn position and shoot direction.

I understand. Your bullet system is really multifunctional and making an editor system will take a really long time... But basic system would be really cool! If you ever want to do it.

Thank you for answering me!

**Comment by Dark-Peace** [comment]
Oh but the preview wouldn't be in a separate editor anyway. It would go against the _node referenced by IDs workflow_ which allows for more flexibility.
